superset/assets/src/dashboard/components/Header.jsx

```diff
@@ -6,12 +6,21 @@ import HeaderActionsDropdown from './HeaderActionsDropdown';
 import EditableTitle from '../../components/EditableTitle';
 import Button from '../../components/Button';
 import FaveStar from '../../components/FaveStar';
+import ObjectTags from '../../components/ObjectTags';
 import UndoRedoKeylisteners from './UndoRedoKeylisteners';
+import {
+  addTag,
+  deleteTag,
+  fetchSuggestions,
+  fetchTags,
+} from '../../welcome/Tags';
 
 import { chartPropShape } from '../util/propShapes';
 import { t } from '../../locales';
 import { UNDO_LIMIT, SAVE_TYPE_OVERWRITE } from '../util/constants';
 
+const CSRF_TOKEN = (document.getElementById('csrf_token') || {}).value;
```

Comment:
I'm not sure about doing this, and less sure about doing this in module scope. If it's not in the redux store already, we could make sure it makes its way through here:
https://github.com/apache/incubator-superset/blob/4bf69a7260f1d4c8fbfe9d9c0300cae39ca3bf81/superset/views/base.py#L93

Reply:
Yeah, I was unsure about this but it's being done in other parts of the code. Let me see if I can fix it.

Reply:
@mistercrunch, I'm not sure how to avoid doing this. It seems to be in line with [the docs for CSRF in Flask](http://flask-wtf.readthedocs.io/en/stable/csrf.html#javascript-requests): the token is only present in the DOM if the user is authenticated (since it's added by the template), so my understanding is that it should be fine. And the token is already there in the DOM at this point, assigning it to `CSRF_TOKEN` here is just a shortcut.

I looked at other AJAX requests in Superset. They're using jQuery after calling `initJQueryAjax`, which also loads the token into a `$.settings` so it can be reused in all subsequent requests.

Reply:
Gotcha, now I'd rather not having it in module scope for the implications it has that the browser exist, which could be an issue while importing during unit tests or for code introspection. Simply importing the file in node might spit out something about `document` not existing.